### PR TITLE
[MAINTENANCE] Migrate `MicrosoftTeamsNotificationAction` to V1

### PR DESCRIPTION
**Pull request by cdkini**
Migrate final action to V1

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/checkpoint/actions.py [hunk 1]
```diff
@@ -549,6 +555,26 @@ def _validate_renderer(cls, renderer: dict | MicrosoftTeamsRenderer) -> Microsof
...
+        if not self._is_enabled(success=success):
+            return {"microsoft_teams_notification_result": None}
+
+        data_docs_pages = self._get_data_docs_pages_from_prior_action(action_context=action_context)
+
+        payload = self.renderer.v1_render(
+            checkpoint_result=checkpoint_result,
+            data_docs_pages=data_docs_pages,
+        )
+        # this will actually sent the POST request to the Microsoft Teams webapp server
```
Comment (billdirks) [inline great_expectations/checkpoint/actions.py:642]: nit: s/sent/send/

I'd remove "actually" also.

File: great_expectations/checkpoint/util.py [hunk 2]
```diff
@@ -95,24 +95,19 @@ def send_opsgenie_alert(query: str, message: str, settings: dict) -> bool:
...
     try:
-        response = session.post(url=microsoft_teams_webhook, json=query)
+        response = session.post(url=microsoft_teams_webhook, json=payload)
+        response.raise_for_status()
     except requests.ConnectionError:
         logger.warning("Failed to connect to Microsoft Teams webhook after 10 retries.")
+        return None
+    except requests.HTTPError as e:
+        logger.warning(f"Request to Microsoft Teams API returned error {response.status_code}: {e}")
+        return None
```
Comment (billdirks) [inline great_expectations/checkpoint/util.py:110]: To have parity with before state you could have an catchall block at the end:
```
except Exception as e:
  logger.error(e)
```

File: tests/actions/test_core_actions.py [hunk 3]
```diff
@@ -462,8 +469,6 @@ def test_MicrosoftTeamsNotificationAction_bad_request(
         validation_result_suite=validation_result_suite,
     ) == {"microsoft_teams_notification_result": None}
 
-    assert "Request to Microsoft Teams webhook returned error 400" in caplog.text
```
Comment (billdirks) [inline tests/actions/test_core_actions.py:468]: Should we still make this assert? It doesn't look like the rest of this method changed.
If we do remove it, you can remove the `caplog` fixture from the arguments to this method and setting the loglevel.

**Review by billdirks — APPROVED** [review]
A few small comments, otherwise LGTM. Thank you!

File: great_expectations/checkpoint/util.py (hunk 2 above)
Comment (cdkini) [reply great_expectations/checkpoint/util.py:110]: Hmm we don't need to maintain parity with existing behavior right? I'm wary of catch-alls like this

**Pull request merged**